app/controllers/api/v1/projects_controller.rb

```diff
@@ -58,13 +58,34 @@ def index
   end
 
   def update
+    primary_language_update = update_params.delete(:primary_language)
     super do |resource|
-      # TODO: extract this primary project content update
-      # to a service object that sits in the project
-      # transaction.
       content_attributes = primary_content_attributes(update_params)
-      unless content_attributes.blank?
-        resource.primary_content.update!(content_attributes)
+
+      if primary_language_update
```

Comment:
This needs an extraction to a service object or class that handles the project_contents update instead of this large conditional behaviour i have here.